app/src/main/java/org/systers/mentorship/view/fragments/EditProfileFragment.kt

```diff
@@ -110,6 +112,7 @@ class EditProfileFragment : DialogFragment() {
             }
             if (currentUser != editProfileBinding.user && errors.isEmpty()) {
                 profileViewModel.updateProfile(editProfileBinding.user!!)
+                preferenceUtils.setUser(editProfileBinding.user!!)
```

Comment:
@Priyansh-Kedia You've marked this as non-null whereas the attributes it takes are nullable so this may lead to the nullability of `user` and result in a crash. Please look into this and revert/change accordingly

Reply:
I checked on this too
We are not getting nullable user from the backend and that is the reason that we would not result in a crash

Reply:
Fine, if this is the case, then please put forward the reason for marking the parameters that `User` takes as nullable

Reply:
SO I introduced nullability so as to introduce a check because we can never be too cautious
This can be made use if some time there is a problem with the data in the backend

Reply:
Stop hit and trial, provide some robust and generalised solution, doing all this is just like running away from responsibilities. 
Consider all the parameters of user entity equally.

Reply:
From the android part of this data, this is a robust solution
I have introduced nullability only *if* the data is null which luckily has not been the case till now due to our backend